### PR TITLE
Improve layout and mobile experience; add opengraph metadata

### PR DESCRIPTION
**Pull request by csauve**
This is a second pass on styles/layout. There's more to do with a sidebar, but future tickets capture that work. For now, I've left it at adding a header bar with placeholder search and some links, and some skeleton markup/styles for a sidebar (not currently visible).

I added a number of metadata properties to the site to aid appearance outside the page. Firstly, re-using the logo image as the favicon and secondly adding opengraph meta tags so we can control how discord creates previews when links to c20 which are pasted in channels.

Lastly I refactored `shared.js` in the `templates` directory by splitting it up into smaller files.

layout

### CODE REVIEW

File: build-config.json [hunk 1]
```diff
@@ -6,10 +6,11 @@
     "srcResources": "./src/content/**/*.@(jpg|jpeg|png|gif)",
     "srcDiagrams": "./src/content/**/*.@(dot|neato|fdp|sfdp|twopi|circo)",
     "srcContentBase": "./src/content",
-    "srcAssetImages": "./src/@(assets)/**/*.@(jpg|jpeg|png|gif)",
+    "srcStaticAssets": "./src/@(assets)/**/*.@(jpg|jpeg|png|gif|ttf)",
```
Comment (csauve) [inline build-config.json:9]: I'm now bundling an Xbox UI font released by Microsoft. It looks nice for headers.

File: build-config.json [hunk 2]
```diff
@@ -6,10 +6,11 @@
...
     "srcStyleEntry": "./src/assets/style.scss",
     "srcStylesAny": "./src/assets/**/*.scss",
     "vendorAssets": ["./node_modules/highlight.js/styles/atom-one-dark.css"],
     "invaderTagDefsBase": "./lib/invader/src/tag/hek/definition/"
-  }
+  },
+  "baseUrl": "https://c20.reclaimers.net"
```
Comment (csauve) [inline build-config.json:15]: The site now needs to know its own base URL to build absolute URLs for opengraph meta tags.

File: gulpfile.js [hunk 3]
```diff
@@ -20,7 +20,7 @@ function clean() {
 //build the stylesheet from SASS
 function assetStyles() {
   return new Promise((resolve, reject) => {
-    sass.render({file: paths.srcStyleEntry}, (err, res) => {
+    sass.render({file: paths.srcStyleEntry, outputStyle: "compressed"}, (err, res) => {
```
Comment (csauve) [inline gulpfile.js:23]: CSS is now minimized for fewer bytes over the wire

File: src/templates/shared/wrapper.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+    <html lang="en">
+      <head>
+        <meta charset="utf-8"/>
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+        <meta http-equiv="X-UA-Compatible" content="ie=edge">
+        <meta property="og:title" content="${page.title}"/>
+        <meta property="og:site_name" content="The Reclaimers Library"/>
+        <meta property="og:type" content="website"/>
+        <meta property="og:locale" content="en_US"/>
+        <meta property="og:url" content="${metaIndex.baseUrl}${page._dirUrl}"/>
```
Comment (csauve) [inline src/templates/shared/wrapper.js:32]: These `og:` properties determine how the page is previewed when shared on sites like Facebook or within Discord channels.

File: src/assets/style.scss [hunk 5]
```diff
@@ -140,103 +260,92 @@ table {
...
+
+  .top-header {
+    .c20-name-short {
+      display: inline-block;
+    }
+    .c20-name-long {
+      display: none;
+    }
+  }
+}
```
Comment (csauve) [inline src/assets/style.scss:351]: Mobile-size styles. We can change the layout of the page and hide some unnecessary elements.

File: src/templates/shared/wrapper.js [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    if (parent) {
+      breadcrumbs.push(parent);
+    }
+  }
+  return html`
+    <!DOCTYPE html>
+    <html lang="en">
+      <head>
+        <meta charset="utf-8"/>
+        <meta name="viewport" content="width=device-width, initial-scale=1.0">
```
Comment (csauve) [inline src/templates/shared/wrapper.js:26]: This makes the page not for ants when viewed on a phone.

File: src/templates/shared/wrapper.js [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+                ${STUB_ALERT}
+              `}
+            </article>
+          </main>
+          <footer class="content-footer">
+            <p>
+              <small>
+                This text is available under the <a href="https://creativecommons.org/licenses/by-sa/3.0/">CC BY-SA 3.0 license</a>
+                •
+                <a href="${srcUrl}">Source</a>
```
Comment (csauve) [inline src/templates/shared/wrapper.js:74]: Added a link to sources on GitHub for the current page

File: src/templates/shared/wrapper.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+                •
+                <a href="#">Go to top</a>
+              </small>
+            </p>
+          </footer>
+        </div>
+      </body>
+    </html>
+  `;
+};
```
Comment (gbMichelle) [inline src/templates/shared/wrapper.js:88]: Here

Comment (csauve) [reply src/templates/shared/wrapper.js:88]: I've aimed to componentize the page structure into separate files under `templates` and `shared`. A UI pass like this adds complexity as you've pointed out. I think there are some good points to further componentize `wrapper.js` by pulling out footer, sidebar, and top bar components. I'll make that change in an upcoming PR which is a continuation of this one.

Comment (gbMichelle) [reply src/templates/shared/wrapper.js:88]: Okay, that sounds good. I would prefer a small issue for that though.

**Pull request merged**